SmartApi/smartWebSocketV2.py

```diff
@@ -186,27 +194,38 @@ def subscribe(self, correlation_id, mode, token_list):
                     "tokenList": token_list
                 }
             }
-
+            if mode == 4:
+                for token in token_list:
+                    if token.get('exchangeType') != 1:
+                        error_message = "Invalid ExchangeType: Please check the exchange type and try again"
+                        logger.error(error_message)
+                        raise ValueError(error_message)
             if self.input_request_dict.get(mode) is None:
                 self.input_request_dict[mode] = {}
-
             for token in token_list:
                 if token['exchangeType'] in self.input_request_dict[mode]:
                     self.input_request_dict[mode][token['exchangeType']].extend(token["tokens"])
                 else:
                     self.input_request_dict[mode][token['exchangeType']] = token["tokens"]
-
             if mode == self.DEPTH:
                 total_tokens = sum(len(token["tokens"]) for token in token_list)
                 quota_limit = 50
                 if total_tokens > quota_limit:
-                    raise Exception("Quota exceeded: You can subscribe to a maximum of {} tokens.".format(quota_limit))
-
-            self.wsapp.send(json.dumps(request_data))
-            self.RESUBSCRIBE_FLAG = True
-
+                    error_message = f"Quota exceeded: You can subscribe to a maximum of {quota_limit} tokens."
+                    logger.error(error_message)
+                    raise Exception(error_message)
+                else:
+                    self.wsapp.send(json.dumps(request_data))
+                    self.RESUBSCRIBE_FLAG = True
         except Exception as e:
-            print("Error:", e)
+            error_message = str(e)
+            if "Quota exceeded" in error_message:
+                logger.error(f"Quota exceeded error: {error_message}")
+                os._exit(1)  # Exit the script when the quota is exceeded
```

Comment:
this should be graceful handle.

Reply:
Remove the os._exit(1) and handle it gracefully